enumerable.rb

```diff
@@ -0,0 +1,124 @@
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    my_array = is_a?(Range) ? to_a : self
+
+    my_array.size.times { |n| yield (my_array[n]) }
+    self
+  end
```

Comment:
### #my_each 
- [ ] when a block is given and self is a hash behaves like #each calls the given block once for each element in self

Reply:
Already works!!!!!!!!!

> # Status: 🔴 Required Changes 🔴
> ## Hey! @brugobi 👋
> In order to get your code to work, we need to make sure that #my_each works as it should. otherwise, all the rest of your function will have the wrong behavior.
> 
> right now when testing your code I get an error when a block is given and self is a hash; instead of receiving the given block once for each element in self, I got an empty string.
> 
> so in order to make it work properly, we need to use 2 different conditions, the first one will check if block is given and the second one will check the method used, otherwise as you already know should return `self` which returns an enum hash. right now you are only checking for the Range method while missing the other
> 
> If you need some extra help you can write to me through slack.
> 
> ### now lest dive in your code.
> ## Please make sure to fix all issues pointed below before submitting a new review.

Please share your test cases